perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/GroupsManagerMethod.java

```diff
@@ -295,21 +297,94 @@ public Group call(ApiCaller ac, Deserializer parms) throws PerunException {
 		}
 	},
 
+
+	/*#
+	 * Adds members to a group. If already a member of the group, the member will be skipped.
+	 * Non-empty list of members expected, if empty, no member will be added.
+	 *
+	 * @param groups List<Group> List of groups that the member should be added to <code>id</code>
+	 * @param member int Member <code>id</code>
+	 */
+	addMembers {
+		@Override
+		public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
+			ac.stateChangingCheck();
+
+			List<Integer> memberInts = parms.readList("members", Integer.class);
+			if (memberInts == null) {
+				throw new RpcException(RpcException.Type.MISSING_VALUE, "Non-empty list of members not sent.");
+			}
+			Group group = ac.getGroupById(parms.readInt("group"));
+			List<Member> members = new ArrayList<>();
+			for (Integer memberInt : memberInts) {
+				Member member = ac.getMemberById(memberInt);
+				if (!ac.getGroupsManager().isGroupMember(ac.getSession(), group, member)) {
```

Comment:
Maybe you could use ``isDirectGroupMember()`` which is currently present only in impl/dao layer (``GroupsManagerImplApi``). Just add such method to the BL and API (entry) layers and then use it here.